### DISCUSSION TITLE
Ensure Space for a Table

### DISCUSSION THREAD

**Discussion by omarraja786** [QuestPDF/QuestPDF#176] [Q&A] [answered]
Hello just another question:

I have a table and I want a certain part to be on a new page if it cannot fit on the single page:

```
 container.PaddingVertical(20).Table(table =>
            {

                table.ColumnsDefinition(columns =>
                {
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                    columns.RelativeColumn();
                });

                table.Cell().Row(1).Column(1).Text($"Quote").Style(titleStyle);
                table.Cell().Row(2).Column(1).Text(Placeholders.Name()).FontSize(8).FontFamily("Verdana").Bold();
                table.Cell().Row(3).Column(1).Text(Placeholders.Name()).FontSize(8).FontFamily("Verdana");

              //Everything from here should be on a new page if all the rows dont fit on the page
               table.Cell().Row(13).Column(1).ColumnSpan(4).PaddingTop(-10).Text("- - - - - - - - -").FontSize(8).FontFamily("Verdana").Bold();


                table.Cell().Row(14).Column(1).Text($"PAYMENT ADVICE").Style(titleStyle);

                table.Cell().Row(15).Column(1).AlignLeft().Text("To:").FontSize(8).FontFamily("Verdana");
                table.Cell().Row(16).Column(1).AlignLeft().Text("Company Name").FontSize(8).FontFamily("Verdana");
                table.Cell().Row(17).Column(1).AlignLeft().Text("Building 50").FontSize(8).FontFamily("Verdana");
}
```

Even if i added (EnsureSpace 500 as an extreme value) it would not take everything from the - - - and go onto a new page.
`table.Cell().Row(13).Column(1).ColumnSpan(4).PaddingTop(-10).EnsureSpace(500).Text("- - - - - - - - -").FontSize(8).FontFamily("Verdana").Bold();`

Is there something im missing? - Note I do want to keep the " - - - " its meant to be an area for the customer to cut out

Thank you
Omar

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁

My recommendation is to use [the Column element](https://www.questpdf.com/documentation/api-reference.html#column) to divide your table into separate sections. Then, you have more granual control where potential brake may happen 😁 To limit repeating code, e.g. defining table columns, you can use local functions like [in this example](https://www.questpdf.com/documentation/api-reference.html#basic-usage).

So, first column item would be "Quote" with the "cut line", and the second item would be the "Payment advice" content (with the EnsureSpace element). Depending on the page complexity, you can also replace the Table element with much [simpler Row element](https://www.questpdf.com/documentation/api-reference.html#row).

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
